akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/GroupRouterImpl.scala

```diff
@@ -71,7 +73,8 @@ private final class InitialGroupRouterImpl[T](
   private val stash = StashBuffer[T](context, capacity = 10000)
 
   def onMessage(msg: T): Behavior[T] = msg match {
-    case serviceKey.Listing(update) =>
+    case serviceKey.Listing(allActorRef) =>
+      val update = if (localAffinity) allActorRef.filter(_.path.address.hasLocalScope) else allActorRef
```

Comment:
How it should work is that having `localAffinity = true` makes the router prefer local routees, but use remote if there are no local.

Reply:
so as raboof said. maybe I can use remote actorref if there is no local actor ref, I will fix it.

Reply:
Yes, exactly 👍

Reply:
Also , note that it already does some come complicated logic here with preferring reachable over all registered actors for the service key, so it will need to be a three step logic with local, if no local reachable, if no reachable unreachable routees.

Reply:
maybe I should dive into the code and think more about it.